app/models/discussion_reader.rb

```diff
@@ -97,12 +97,7 @@ def viewed!(age_of_last_read_item = nil)
       reset_counts!
     end
 
-    publish!
-  end
-
-  def publish!
-    publish_data = Simple::DiscussionSerializer.new(discussion, scope: { reader_cache: reader_cache })
-    MessageChannelService.publish(publish_data.as_json, to: user)
+    EventBus.broadcast('discussion_reader_viewed', discussion, self, user)
```

Comment:
So far the EventBus hasn't needed to dip down to the model level (it's called entirely from services). Is there a reason we can't call this from DiscussionService#mark_as_read?